src/ussd_sassa_registration.js

```diff
@@ -67,109 +93,389 @@ go.app = (function () {
+                      return $(error_msg);
+                    } else if (response.data.existing) {
+                      return $(
+                        [
+                          "An application for {{ id_number }} has already been made. The status is: {{ status }}.",
+                          "To appeal, call 0800 002 9999",
+                        ].join("\n")
+                      ).context({
+                        id_number: content,
+                        status: response.data.status,
+                      });
+                    }
                   }
-                  return self.states.create(name, opts);
-                }
-              );
+                );
+            }
           }
         },
         next: "state_grant",
       });
     });
 
+    self.add("state_grant", function (name) {
+      return new ChoiceState(name, {
+        question: $(
+          "Do you (or the applicant) receive a child grant or social grant from SASSA?"
+        ),
+        error: $(
+          [
+            "Sorry, I don't understand that answer. Please choose one of the options below.",
+            "Do you receive a child grant or social grant from SASSA?",
+          ].join("\n")
+        ),
+        accept_labels: true,
+        choices: [new Choice(true, $("Yes")), new Choice(false, $("No"))],
+        next: "state_uif",
+      });
+    });
+
+    self.add("state_uif", function (name) {
+      return new ChoiceState(name, {
+        question: $(
+          "Do you (or the applicant) receive, or have you applied for, " +
+            "Unemployment Insurance Fund (UIF) benefits?"
+        ),
+        error: $(
+          [
+            "Sorry, I don't understand that answer. Please choose one of the options below.",
+            "",
+            "Do you receive, or have you applied for UIF benefits?",
+          ].join("\n")
+        ),
+        accept_labels: true,
+        choices: [new Choice(true, $("Yes")), new Choice(false, $("No"))],
+        next: "state_income",
+      });
+    });
+
+    self.add("state_income", function (name) {
+      return new ChoiceState(name, {
+        question: $(
+          "Do you (or the applicant) currently receive any income from any " +
+            "other source – salary, wages, private or civil pension or annuity?"
+        ),
+        error: $(
+          "Do you currently receive any income from any other source – salary, " +
+            "wages, private or civil pension or annuity?"
+        ),
+        accept_labels: true,
+        choices: [new Choice(true, $("Yes")), new Choice(false, $("No"))],
+        next: "state_name",
+      });
+    });
+
+    self.add("state_name", function (name) {
+      return new FreeText(name, {
+        question: $(
+          "Please enter the applicant's name and surname as per identity " +
+            "document (eg John Doe)"
+        ),
+        check: function (content) {
+          if (content.length < 2 || content.indexOf(" ") == -1) {
```

Comment:
Maybe we should do a `.trim()` before checking length, so that two spaces doesn't match? Or maybe a regex of `\S+\s\S+.*`

Reply:
We should probably also do a length limitation or truncation, so that we don't hit our db field limit, or the character limit for the confirm screen